### ISSUE TITLE
The new resolver cannot reference properties of JavaScript variables

### ISSUE THREAD

**Issue by pizzacat83** [serverless/serverless#9155]
As described in the documentation, the old resolver can resolve properties of JavaScript variables.

```yaml
# serverless.yml
service: new-service
provider: aws
functions:
  scheduledFunction:
    handler: handler.scheduledFunction
    events:
      - schedule: ${file(../myCustomFile.js):schedule.ten}
```

```js
// myCustomFile.js
module.exports.schedule = () => {
  // Code that generates dynamic data
  return {
    ten: 'rate(10 minutes)',
    twenty: 'rate(20 minutes)',
    thirty: 'rate(30 minutes)',
  };
};
```
(https://www.serverless.com/framework/docs/providers/aws/guide/variables/#reference-variables-in-javascript-files)

However, the new resolver treats `schedule.ten` as `module.exports['schedule.ten']`, not `module.exports.schedule.ten`. Thus, it cannot reference properties of exported objects.

<details>
<summary><code>serverless.yml</code> (old resolver)</summary>

```yaml
service: myservice

custom:
  foobar: '${file(foobar.js):foo.bar}'
  foobarFunc: '${file(foobarFunc.js):foo.bar}'

provider: aws
```
</details>

<details>
<summary><code>serverless.yml</code> (new resolver)</summary>

```yaml
service: myservice

variablesResolutionMode: 20210219 # this is the only difference between `serverless.yml (old resolver)`

custom:
  foobar: '${file(foobar.js):foo.bar}'
  foobarFunc: '${file(foobarFunc.js):foo.bar}'

provider: aws
```

</details>

<details>
<summary><code>foobar.js</code></summary>

```js
module.exports.foo = {
  bar: "exports.foo.bar"
};

module.exports['foo.bar'] = "exports['foo.bar']";
```

</details>

<details>
<summary><code>foobarFunc.js</code></summary>

```js
module.exports.foo = () => ({
  bar: "exports.foo.bar"
});

module.exports['foo.bar'] = () => "exports['foo.bar']";
```

</details>

<details>
<summary><b><code>npx sls print</code> output (old resolver)</b></summary>

```
Serverless: Deprecation warning: Starting with next major version, default value of provider.lambdaHashingVersion will be equal to "20201221"
            More Info: https://www.serverless.com/framework/docs/deprecations/#LAMBDA_HASHING_VERSION_V2
service:
  name: myresolver
custom:
  foobar: exports.foo.bar
  foobarFunc: exports.foo.bar
provider:
  stage: dev
  region: us-east-1
  name: aws
  versionFunctions: true
  variableSyntax: \${([^{}:]+?(?:\(|:)(?:[^:{}][^{}]*?)?)}

```

</details>

<details>
<summary><b><code>npx sls print</code> output (new resolver)</b></summary>

```
service:
  name: myresolver
variablesResolutionMode: '20210219'
custom:
  foobar: exports['foo.bar']
  foobarFunc: exports['foo.bar']
provider:
  stage: dev
  region: us-east-1
  name: aws
  versionFunctions: true
  variableSyntax: \${([^{}:]+?(?:\(|:)(?:[^:{}][^{}]*?)?)}
```

</details>

<b>Installed version (old resolver)</b>

```
Framework Core: 2.25.2 (local)
Plugin: 4.5.1
SDK: n/a
Components: 3.7.5
```

<b>Installed version (new resolver)</b>

```
Framework Core: 2.30.3 (local)
Plugin: 4.5.1
SDK: 4.2.1
Components: 3.7.5
```

**Comment by pgrzesik** [comment]
Thanks for reporting @pizzacat83 :bow: @medikoo Would you be able to take a look at this report?

**Comment by medikoo** [comment]
@pizzacat83 great thanks for reporting.

You've actually discovered two issues:
1. Variables function resolver was configured to run resolved functions only if they were found at final target properties. so in your case, resolver was resolving `ten` property on `schedule` function (without invoking it). It'll attempt to invoke function at `ten` property, bu totally ignored the fact that it's `schedule` that could be a function.
2. To resolve address we've relied on lodash's `get`, and in case of ambiguity in form that there's (1) `foo` object with `bar` property and (2) `foo.bar` property. On `foo.bar` address it was going with the latter `foo.bar` property.

Fixed both issues in https://github.com/serverless/serverless/pull/9161
